build.sh

```diff
@@ -2,30 +2,43 @@
 
 set -e
 
-cd "$(dirname "$(readlink -f "${BASH_SOURCE[0]}")")"
+if [ -z "$1" ]; then
```

Comment:
Since we are already committed to using `bash`, might as well use `[[` over `[`.

Also I'm kind of a person that prefers newlines over `;` but here I don't really care in this case.

```suggestion
if [[ -z "$1" ]]
then
```

Same applies to other `if`s below.